src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -1,33 +1,58 @@
 /* ==================================================
- * Authors:
+ * Authors: Jason Wang
  *
  * --------------------------------------------------
- * Description:
- *
+ * Description: Shooter.java contains the functions to control the shooter.
  * ================================================== */
 
 package frc.robot.subsystems;
 
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.motorcontrol.PWMTalonSRX;
 
 public class Shooter extends SubsystemBase
 {
   // HARDWARE //
-
-  // declare motor controllers, solenoids, and other hardware here
-
-  // CONSTANTS //
-
-
-
-  // CONSTRUCTOR //
-
+  private PWMTalonSRX motor;
+  private double speed;
+  private double motorSpeed;
+ // Ports //
+ public static final int SHOOTER_MOTOR = -1;
+ // Constants // 
   public Shooter()
   {
     // instantiate hardware
+    motor = new PWMTalonSRX(SHOOTER_MOTOR);
+    speed = 0; //default speed
   }
-
   // METHODS //
-
+  // spins the motor for the shooter
+  public void spin(){
+    motor.set(speed);
+  }//stops the motor for the shooter
+  public void stop(){
+    motor.set(0);
+  }//updates the speed for the motor to accomodate the distance from the target
+  // this updated speed would be the speed the algorithm calculates in vision.java
+  public void setSpeed(){
+    this.speed = motorSpeed;
+    if(speed>1){
+      speed=1;
+    }if(speed<0){
+      speed=0;
+    }
```

Comment:
You should pass in the speed as a parameter to the setSpeed() method:
```
setSpeed(double speed)
{
    this.speed = speed;
}

Reply:
Should also probably check to make sure `speed` is not negative (only 0 to 1 allowed)